source/clear-linux/concepts/stateless.rst

```diff
@@ -0,0 +1,145 @@
+.. _stateless:
+
+Stateless
+#########
+
+In most operating systems, files can become intermingled with user and system
+data and configurations.
+
+.. figure:: figures/stateless-1.png
+   :scale: 45% 
+   :align: center
+   :alt: Stateless: User and system files mixed
+
+   Figure 1: Without stateless, user and system files become mixed on the filesystem over time.
+
+|CL-ATTR| has a stateless design philosophy of which the goal is to provide an
+:abbr:`OS (operating system)` that functions without excessive user
+configuration or customization. Stateless in this context does *not* mean
+ephemeral or non-persistent.
+
+.. contents:: :local:
+   :depth: 2
+
+
+File-level separation
+*********************
+
+To accomplish a stateless design the Linux Filesystem Hierarchy is separated
+between user-owned areas and |CL|-owned areas.
+
+.. figure:: figures/stateless-2.png
+   :scale: 45% 
+   :align: center
+   :alt: Stateless: User and system files separation
+
+   Figure 2: With stateless, user and system files are separated on the filesystem.
+
+System areas
+============
+File under the :file:`/usr` directory are managed by |CL| as system files.
+Files written under the :file:`/usr` directory by users can get removed
+through system updates with :ref:`swupd <swupd-about>`.This operating
+assumption allows |CL| to verify and maintain integrity of system files.
+
+User areas
+==========
+Files under the :file:`/etc/`, :file:`/home`, and :file:`/var` directories are
+owned and managed by the user. A freshly installed |CL| system will only have
+a minimal set of files in the :file:`/etc/` directory and software installed
+by |CL| does not write to :file:`/etc`. This operating assumption allows |CL|
+users to clearly identify the configuration that makes their system unique. 
+
+
+Software Configuration
+**********************
+
+With stateless separation, default software configurations are read in order
+from predefined source code, |CL| provided defaults, and user-provided
+configuration.
+
+Default configurations
+======================
+
+Software in |CL| provides default configuration values so that it is
+immediately functional, whenever it is appropriate to do so. 
+
+|CL| distributed software packages may be directly modified to include default
+configuration values or default configuration files may be provided by |CL|
+under :file:`/usr/share/defaults`. These files can be referenced as templates
+for customization. 
+
+For example, the default configuration that Apache uses when installed can be
+found at :file:`/usr/share/defaults/httpd/httpd.conf` directory.
+
+
+Overriding configurations
+=========================
+
+If a configuration needs to be changed, the appropriate file should be
+modified by the user under :file:`/etc/`. If the configuration file does not
+already exist, it can be created in the appropriate location.
+
+User defined configuration files should contain the minimal set of desired
```

Comment:
>User defined configuration files should contain the minimal set of desired
changes and rely on default configuration for the rest. 

@ahkok I'm particularly interested if you are in agreement with this statement for CL..

Reply:
Pretty sure I wrote this.